pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DistinctFromTransformFunction.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.roaringbitmap.RoaringBitmap;
+
+
+/**
+ * <code>DistinctFromTransformFunction</code> abstracts the transform needed for IsDistinctFrom and IsNotDistinctFrom.
+ * Null value is considered as distinct from non-null value.
+ * When both values are not null, this function calls equal transform function to determined whether two values are
+ * distinct.
+ * This function only supports two arguments which are both column names.
+ */
+public class DistinctFromTransformFunction extends BinaryOperatorTransformFunction {
+  // If _getDistinct is true, act as IsDistinctFrom transformation, otherwise act as IsNotDistinctFrom.
+  private boolean _getDistinct;
+
+  /**
+   * @param getDistinct is set to true for IsDistinctFrom, otherwise it is for IsNotDistinctFrom.
+   */
+  protected DistinctFromTransformFunction(boolean getDistinct) {
+    super(getDistinct ? TransformFunctionType.NOT_EQUALS : TransformFunctionType.EQUALS);
+    _getDistinct = getDistinct;
+  }
+
+  /**
+   * Returns a bit map of corresponding column.
+   * Returns an empty bitmap by default if null option is disabled.
+   */
+  private static RoaringBitmap getNullBitMap(ProjectionBlock projectionBlock, TransformFunction transformFunction) {
+    String columnName = ((IdentifierTransformFunction) transformFunction).getColumnName();
+    RoaringBitmap nullBitmap = projectionBlock.getBlockValueSet(columnName).getNullBitmap();
+    if (nullBitmap == null) {
+      return new RoaringBitmap();
+    }
+    return nullBitmap;
+  }
+
+  @Override
+  public String getName() {
+    if (_getDistinct) {
+      return TransformFunctionType.IS_DISTINCT_FROM.getName();
+    }
+    return TransformFunctionType.IS_NOT_DISTINCT_FROM.getName();
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    super.init(arguments, dataSourceMap);
+    if (!(_leftTransformFunction instanceof IdentifierTransformFunction)
+        || !(_rightTransformFunction instanceof IdentifierTransformFunction)) {
+      throw new IllegalArgumentException("Only column names are supported in DistinctFrom transformation.");
+    }
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return BOOLEAN_SV_NO_DICTIONARY_METADATA;
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
+    int length = projectionBlock.getNumDocs();
+    _results = super.transformToIntValuesSV(projectionBlock);
+    RoaringBitmap letNull = getNullBitMap(projectionBlock, _leftTransformFunction);
+    RoaringBitmap rightNull = getNullBitMap(projectionBlock, _rightTransformFunction);
+    RoaringBitmap xorNull = RoaringBitmap.xor(letNull, rightNull);
+    for (int i = 0; i < length; i++) {
+      if (xorNull.contains(i)) {
+        // These are docs that are null one side and not null the other side.
+        _results[i] = _getDistinct ? 1 : 0;
```

Comment:
For better performance, we may have an int value as the member variable. (Don't remember if this is the reason why you use int initially for `_getDistinct`)

Reply:
Done